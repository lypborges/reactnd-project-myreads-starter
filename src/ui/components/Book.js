import React, { Component } from "react";
import BookShelfChanger from "./BookShelfChanger";
import PropTypes from "prop-types";

class Book extends Component {
  static propTypes = {
    book: PropTypes.shape({
      title: PropTypes.string.isRequired,
      authors: PropTypes.array.isRequired,
      imageLinks: PropTypes.object.isRequired
    })
  };

  render(props) {
    const { book } = this.props;
    return (
      <div className="book">
        <div className="book-top">
          <div
            className="book-cover"
            style={{
              width: 128,
              height: 193,
              backgroundImage: `url(${book.imageLinks.thumbnail})`
            }}
          />
          <BookShelfChanger />
        </div>
        <div className="book-title">{book.title}</div>
        {book.authors.map(author => {
          return (
            <div className="book-authors" key={author}>
              {author}
            </div>
          );
        })}
      </div>
    );
  }
}

export default Book;
