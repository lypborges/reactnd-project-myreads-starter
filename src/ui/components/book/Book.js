import React, { Component } from "react";
import PropTypes from "prop-types";
import { Card, Radio } from "antd";
import Author from "../author/Author";

const { Meta } = Card;

class Book extends Component {
  static propTypes = {
    book: PropTypes.shape({
      title: PropTypes.string,
      authors: PropTypes.array,
      shelf: PropTypes.string,
      imageLinks: PropTypes.object
    })
  };

  handleChange = e => {
    this.props.handleShelfChange(e.target.value, this.props.book);
  };

  render(props) {
    if (!this.props.book) return null;

    const bookStyle = {
      width: 248,
      height: 193
    };
    const { book } = this.props;
    const { imageLinks, authors, shelf, title } = book;
    return (
      <div className="bookshelf-container">
        <Card
          className="bookshelf-book"
          style={{ width: 250 }}
          cover={
            imageLinks ? (
              <img alt={title} style={bookStyle} src={imageLinks.thumbnail} />
            ) : (
              <img
                alt={title}
                style={bookStyle}
                src="http://via.placeholder.com/248x193"
              />
            )
          }
        >
          <Meta title={title} description={<Author authors={authors} />} />
        </Card>
        <div>
          <Radio.Group
            value={shelf}
            size={"small"}
            onChange={this.handleChange}
          >
            <Radio.Button
              data-testid="button-currently-reading"
              value="currentlyReading"
            >
              Reading
            </Radio.Button>
            <Radio.Button data-testid="button-want-to-read" value="wantToRead">
              Want to Read
            </Radio.Button>
            <Radio.Button data-testid="button-read" value="read">
              Read
            </Radio.Button>
            <Radio.Button data-testid="button-none" value="none">
              None
            </Radio.Button>
          </Radio.Group>
        </div>
      </div>
    );
  }
}

export default Book;
