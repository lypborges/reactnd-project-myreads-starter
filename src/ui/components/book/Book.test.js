import React from "react";
import renderer from "react-test-renderer";
import Book from "./Book";
import { BookData } from "../../../tests/helpers/BookStore";

it("renders correctly with book", () => {
  const tree = renderer.create(<Book book={BookData} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly without book", () => {
  const tree = renderer.create(<Book />).toJSON();
  expect(tree).toMatchSnapshot();
});
