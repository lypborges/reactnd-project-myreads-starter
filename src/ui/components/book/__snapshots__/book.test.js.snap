// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with book 1`] = `
<div
  className="bookshelf-container"
>
  <div
    className="ant-card bookshelf-book ant-card-bordered"
    style={
      Object {
        "width": 250,
      }
    }
  >
    <div
      className="ant-card-cover"
    >
      <img
        alt="The Linux Command Line"
        src="http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
        style={
          Object {
            "height": 193,
            "width": 248,
          }
        }
      />
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="ant-card-meta"
      >
        <div
          className="ant-card-meta-detail"
        >
          <div
            className="ant-card-meta-title"
          >
            The Linux Command Line
          </div>
          <div
            className="ant-card-meta-description"
          >
            <span>
              William E. Shotts, Jr.
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="ant-radio-group ant-radio-group-small"
      id={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      style={undefined}
    >
      <label
        className="ant-radio-button-wrapper ant-radio-button-wrapper-checked"
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        style={undefined}
      >
        <span
          className="ant-radio-button ant-radio-button-checked"
          style={Object {}}
        >
          <input
            autoFocus={undefined}
            checked={true}
            className="ant-radio-button-input"
            data-testid="button-currently-reading"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={undefined}
            onFocus={[Function]}
            readOnly={undefined}
            tabIndex={undefined}
            type="radio"
            value="currentlyReading"
          />
          <span
            className="ant-radio-button-inner"
          />
        </span>
        <span>
          Reading
        </span>
      </label>
      <label
        className="ant-radio-button-wrapper"
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        style={undefined}
      >
        <span
          className="ant-radio-button"
          style={Object {}}
        >
          <input
            autoFocus={undefined}
            checked={false}
            className="ant-radio-button-input"
            data-testid="button-want-to-read"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={undefined}
            onFocus={[Function]}
            readOnly={undefined}
            tabIndex={undefined}
            type="radio"
            value="wantToRead"
          />
          <span
            className="ant-radio-button-inner"
          />
        </span>
        <span>
          Want to Read
        </span>
      </label>
      <label
        className="ant-radio-button-wrapper"
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        style={undefined}
      >
        <span
          className="ant-radio-button"
          style={Object {}}
        >
          <input
            autoFocus={undefined}
            checked={false}
            className="ant-radio-button-input"
            data-testid="button-read"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={undefined}
            onFocus={[Function]}
            readOnly={undefined}
            tabIndex={undefined}
            type="radio"
            value="read"
          />
          <span
            className="ant-radio-button-inner"
          />
        </span>
        <span>
          Read
        </span>
      </label>
      <label
        className="ant-radio-button-wrapper"
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        style={undefined}
      >
        <span
          className="ant-radio-button"
          style={Object {}}
        >
          <input
            autoFocus={undefined}
            checked={false}
            className="ant-radio-button-input"
            data-testid="button-none"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={undefined}
            onFocus={[Function]}
            readOnly={undefined}
            tabIndex={undefined}
            type="radio"
            value="none"
          />
          <span
            className="ant-radio-button-inner"
          />
        </span>
        <span>
          None
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`renders correctly without book 1`] = `null`;
